src/analysis/inner_product_space/lax_milgram.lean

```diff
@@ -0,0 +1,178 @@
+  rw lax_milgram_map_apply,
+  exact is_lax_milgram w,
+end
+
+variables {B}
+
+lemma antilipschitz_of_lax_milgram (coercive : is_coercive B) :
+  ∃ C : ℝ, 0 < C ∧ antilipschitz_with C.to_nnreal (lax_milgram_map B) :=
+begin
+  rcases coercive with ⟨C, C_pos, coercivity⟩,
+  refine ⟨C⁻¹, inv_pos.mpr C_pos, _⟩,
+  refine linear_map.antilipschitz_of_bound ↑(lax_milgram_map B) _,
+  intros v,
+  by_cases h : 0 < ∥v∥,
+  { rw [←mul_le_mul_left C_pos, ←mul_assoc],
+    simp only [real.coe_to_nnreal', normed_group.dist_eq, max_eq_left_of_lt (inv_pos.mpr C_pos)],
+    rw [mul_inv_cancel, one_mul],
+    show C ≠ 0, by linarith,
+    refine (mul_le_mul_right h).mp _,
+    exact calc
+         C * ∥v∥ * ∥v∥
+         ≤ B v v                         : coercivity v
+    ...  = inner (lax_milgram_map B v) v : by simp
+    ...  ≤ ∥lax_milgram_map B v ∥ * ∥v∥    : real_inner_le_norm (lax_milgram_map B v) v, },
+  { have : v = 0 := by simpa using h,
+    simp [this], }
+end
+
+lemma injective (coercive : is_coercive B) : (lax_milgram_map B).ker = ⊥ :=
+begin
+  simp only [submodule.eq_bot_iff, continuous_linear_map.mem_ker],
+  intros v zero_of_image,
+  rw ←norm_le_zero_iff,
+  rcases antilipschitz_of_lax_milgram coercive with ⟨C, C_pos, antilipschitz⟩,
+  have :=  antilipschitz v 0,
+  simpa [edist_dist, zero_of_image, norm_zero] using this,
+end
+
+lemma closed_range (coercive : is_coercive B) : is_closed ((lax_milgram_map B).range : set V) :=
+begin
+  rcases antilipschitz_of_lax_milgram coercive with ⟨_, _, antilipschitz⟩,
+  exact antilipschitz.is_closed_range (lax_milgram_map B).uniform_continuous,
+end
+
+lemma surjective (coercive : is_coercive B): (lax_milgram_map B).range = ⊤ :=
+begin
+  haveI := (closed_range coercive).complete_space_coe,
+  rw ← (lax_milgram_map B).range.orthogonal_orthogonal,
+  rw submodule.eq_top_iff',
+  intros v w mem_w_orthogonal,
+  rcases coercive with ⟨C, C_ge_0, coercivity⟩,
+  have : C * ∥w∥ * ∥w∥ ≤ 0 :=
+  calc C * ∥w∥ * ∥w∥
+        ≤ B w w                         : coercivity w
+  ...  = inner (lax_milgram_map B w) w  : by simp
+  ...  = 0                              : mem_w_orthogonal _ ⟨w, rfl⟩,
+  have : ∥w∥ * ∥w∥ ≤ 0 := by nlinarith,
+  have h : ∥w∥ = 0 := by nlinarith [norm_nonneg w],
+  have w_eq_zero : w = 0 := by simpa using h,
+  simp [w_eq_zero],
+end
+
+/--
+The Lax-Milgram equivalence of a bounded bilinear operator: for all `v : V`
+`lax_milgram_equiv B v` is the unique element `V`
+such that `⟪lax_milgram_map B v, w⟫ = B v w`.
+The Lax-Milgram theorem states that this is an equivalence.
+-/
+def lax_milgram_equiv (coercive : is_coercive B) : V ≃L[ℝ] V :=
+continuous_linear_equiv.of_bijective
+  (lax_milgram_map B)
+  (injective coercive)
+  (surjective coercive)
+
+
+@[simp]
+lemma lax_milgram_equiv_apply (coercive : is_coercive B) (v w : V) :
+  inner (lax_milgram_equiv coercive v) w = B v w :=
+begin
+  unfold lax_milgram_equiv continuous_linear_equiv.of_bijective lax_milgram_map,
+  simp,
+end
```

Comment:
When you have a situation like this, writing out again the properties of an "upgraded" map that you already knew for the original map, it usually works just to copy out the original lemma names.  So here (untested, but please try it)
```suggestion
lax_milgram_map_apply B v w
```
Likewise for `unique_lax_milgram_equiv`.

Reply:
Works, thank you!